src/libraries/System.Net.Requests/src/System/Net/HttpWebRequest.cs

```diff
@@ -1660,7 +1668,24 @@ private static HttpClient CreateHttpClient(HttpClientParameters parameters, Http
 
                     try
                     {
-                        socket.NoDelay = true;
+                        if (parameters.ServicePoint is not null)
+                        {
+                            socket.NoDelay = !parameters.ServicePoint.UseNagleAlgorithm;
+                            socket.ReceiveBufferSize = parameters.ServicePoint.ReceiveBufferSize;
```

Comment:
The behavior of both UseNagleAlgorithm and ReceiveBufferSize is slightly different in framework than what you have here, see https://referencesource.microsoft.com/#System/net/System/Net/ServicePoint.cs,319:
```c#
            // make this configurable from the user:
            if (!UseNagleAlgorithm) {
                finalSocket.SetSocketOption(SocketOptionLevel.Tcp, SocketOptionName.NoDelay, 1);
            }
            if (ReceiveBufferSize != -1) {
                finalSocket.SetSocketOption(SocketOptionLevel.Socket, SocketOptionName.ReceiveBuffer, ReceiveBufferSize);
            }
```

Also without this condition, `socket.ReceiveBufferSize` will throw if user didn't set it, as the default is -1.

Reply:
But I do agree with Tomas about Nagle, so maybe we should just ignore `UseNagleAlgorithm` or make sure we use it's `true` value only if it was explicitly set by the user and passed in as an unchanged default.

BTW, did any of the customers complained specifically about UseNagleAlgorithm?

Reply:
No, there is no customer who is complaining about the `UseNagleAlgorithm`. I'll remove it with the next commit.